src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package org.folio.consortia.service.impl;
+
+import java.util.UUID;
+
+import org.folio.consortia.domain.dto.SharingInstance;
+import org.folio.consortia.domain.entity.SharingInstanceEntity;
+import org.folio.consortia.exception.ResourceNotFoundException;
+import org.folio.consortia.repository.SharingInstanceRepository;
+import org.folio.consortia.service.ConsortiumService;
+import org.folio.consortia.service.SharingInstanceService;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.stereotype.Service;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class SharingInstanceServiceImpl implements SharingInstanceService {
+
+  private final SharingInstanceRepository sharingInstanceRepository;
+  private final ConsortiumService consortiumService;
+  private final ConversionService converter;
+
+  @Override
+  public SharingInstance getById(UUID consortiumId, UUID actionId) {
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
+    SharingInstanceEntity sharingInstanceEntity = sharingInstanceRepository.findById(actionId).
+      orElseThrow(() -> new ResourceNotFoundException("actionId", String.valueOf(actionId)));
+    return converter.convert(sharingInstanceEntity, SharingInstance.class);
+  }
+
+  @Override
+  @Transactional
+  public SharingInstance save(UUID consortiumId, SharingInstance sharingInstance) {
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
```

Comment:
lets add check here that both tenants from payload exist in DB

Reply:
added

Reply:
We also have following requirement: _Non-central tenants should not be allowed to invoke this endpoint_

Reply:
It is alredy supported, in our case whe have consortium record only in the central tenant, so when someone tryis to invoke this enpoint for some other member tenant - consortium not found error will be thrown that is acceptable